wgengine/monitor/monitor_linux.go

```diff
@@ -35,6 +35,12 @@ type nlConn struct {
 	logf     logger.Logf
 	conn     *netlink.Conn
 	buffered []netlink.Message
+
+	// addrCache is used to suppress duplicate RTM_NEWADDR messages, it is
+	// populated by RTM_NEWADDR messages and de-populated by RTM_DELADDR.
+	// Address change notifications will only passed up to the callees if the
+	// address is not a member of this set. See #4282.
+	addrCache map[string]map[netaddr.IP]struct{}
```

Comment:
whenever you have non-obvious map keys like strings, document what the key is in a comment

also, I'm kinda anti-struct{} for sets. I just use bool true. It makes set membership tests so much cleaner to write.

Reply:
👍